core/shared/src/main/scala/org/http4s/Uri.scala

```diff
@@ -349,7 +349,7 @@ object Uri extends UriPlatform {
       if (endsWithSlash && segments.nonEmpty) writer << "/" else writer
     }
 
-    override val renderString: String = super.renderString
+    override lazy val renderString: String = super.renderString
```

Comment:
What purpose is an override to super serving?

Reply:
Made a `def` a `val`? (this seems moot). And also once it's there, for binary compatibility, although that's not relevant on main.

Reply:
It prevents it from creating a StringWriter on repeated references.  That whole thing deserves a rethink at some point.